h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1161,7 +1161,7 @@ public void init(boolean expensive) {
     }
     Frame tr = _train != null?_train:_parms.train();
     if( tr == null ) { error("_train", "Missing training frame: "+_parms._train); return; }
-    setTrain(new Frame(null /* not putting this into KV */, tr._names.clone(), tr.vecs().clone()));
+    setTrain(new Frame(tr._key, tr._names.clone(), tr.vecs().clone()));
```

Comment:
This is not a correct fix, everything should work with a key-less Frame, the Training frame should never be itself installed in DKV. The original Frame should also never be touched during model building.